### DISCUSSION TITLE
iOS port

### DISCUSSION THREAD

**Discussion by LoveGaming2much** [skyline-emu/skyline#387] [General]
Can someone please port Skyline to ios ?  There is ALTSTORE for sideloading on ios and it can enable jit on non-jailbroken devices.

**Comment by resendex** [comment]
Yeah, it is possible to do this. I myself have the M1 iPad Pro (non jailbroken) and was able to run DolphiniOS on the latest iPadOS 15.4 thanks to AltStore and to a Windows PowerShell Developer Debug workaround. So I'm sure if in the future you really can create an iOS port like Dolphin team did, regardless Apple's tight software restrictions, there's always a way to make it work. :)

**Reply by LoveGaming2much** [reply]
Most devs aren’t interested in developing for iOS. I was thinking about paying the known devs who usually port emus to port Citra, Skyline and or EGG NS (which deserves to be opened and ported).

**Reply by josesilveiraa** [reply]
It's because iOS isn't good. cope, seethe, and cry about it

**Reply by LoveGaming2much** [reply]
iOS is actually very good.

**Reply by 1026251083** [reply]
I don't think anyone wants to rewrite the code

**Comment by iniMerchant64** [comment]
As far as i can tell none of the devs are interested in making an ios port. It might happen but not anytime soon unless someone else decides on making a port

**Comment by PixelyIon** [comment]
There are several hardware limitations that make this non-trivial. There's also the fact that Apple dislikes industry-standard graphics APIs like Vulkan and it would require a Metal backend for the entire GPU which is unlikely to be done at any point. 

We would work on an iOS version if it didn't take this amount of time and investment but it will so it's not in our future plans at the moment.

**Reply by LoveGaming2much** [reply]
I understand now.

**Reply by resendex** [reply]
> There are several hardware limitations that make this non-trivial. There's also the fact that Apple dislikes industry-standard graphics APIs like Vulkan and it would require a Metal backend for the entire GPU which is unlikely to be done at any point. 
> 
> We would work on an iOS version if it didn't take this amount of time and investment but it will so it's not in our future plans at the moment.

it does? so how do you explain this?! been running in Vulkan since ever and it works. [P_20220214_124830.jpg](https://user-images.githubusercontent.com/80842897/153867370-6dc3e213-4ed4-4d4f-991f-9ff38a6cdf1d.jpg)

**Reply by nickbeth** [reply]
That's MoltenVK and the reason why that wouldn't work for Skyline have already been said in this discussion, read further down.

**Reply by resendex** [reply]
oh okay! thanks 🙂

**Comment by LoveGaming2much** [comment]
Would MoltenVK help ?

**Reply by PixelyIon** [reply]
No, it's too limited and you end up running into limitations with it pretty soon. You need a Metal graphics backend for it to work.

**Reply by LoveGaming2much** [reply]
Thank you for the information. I understand why people don’t like porting to iOS. I truly believe that the hardware is amazingly performant but Apple restricts the Os way too much.

**Reply by PixelyIon** [reply]
I entirely agree that it's some great HW and if a port was possible without a lot of effort I'd do it but alas Apple restricts them too much in SW.

**Reply by LoveGaming2much** [reply]
Since Skyline is open-source, a team like the one that ported Dolphin to iOS could take care of porting it.

**Reply by PixelyIon** [reply]
I would prefer if it was official at that point, if any contributor does want to achieve that at some point (not anytime soon though), they can.

**Comment by davyson123** [comment]
hi mark good night, mark would like to know about the mediatek helio g85 processors, if in case it will run games because only give black screen you know. thanks.
